Index: texmaker-1.9/spellerdialog.h
===================================================================
--- texmaker-1.9.orig/spellerdialog.h
+++ texmaker-1.9/spellerdialog.h
@@ -13,7 +13,7 @@
 
 #include "ui_spellerdialog.h"
 #include "latexeditor.h"
-#include "hunspell/hunspell.hxx"
+#include <hunspell.hxx>
 
 #include <QTextCursor>
 
Index: texmaker-1.9/texmaker.pro
===================================================================
--- texmaker-1.9.orig/texmaker.pro
+++ texmaker-1.9/texmaker.pro
@@ -3,7 +3,8 @@ TEMPLATE	= app
 TARGET	 = texmaker
 QT += network \
       xml
-CONFIG	+= qt warn_off release
+CONFIG	+= qt warn_off release link_pkgconfig
+PKGCONFIG = hunspell
 ###############################
 HEADERS	+= texmaker.h \
 	texmakerapp.h \
@@ -39,25 +40,6 @@ HEADERS	+= texmaker.h \
 	manhattanstyle.h \
 	stylehelper.h \
 	styleanimator.h \
-	hunspell/affentry.hxx \
-	hunspell/affixmgr.hxx \
-	hunspell/atypes.hxx \
-	hunspell/baseaffix.hxx \
-	hunspell/csutil.hxx \
-	hunspell/dictmgr.hxx \
-	hunspell/hashmgr.hxx \
-	hunspell/htypes.hxx \
-	hunspell/hunspell.hxx \
-	hunspell/hunspell.h \
-	hunspell/langnum.hxx \
-	hunspell/license.hunspell \
-	hunspell/phonet.hxx \
-	hunspell/suggestmgr.hxx \
-	hunspell/license.myspell \
-	hunspell/filemgr.hxx \
-	hunspell/hunzip.hxx \
-	hunspell/replist.hxx \
-	hunspell/w_char.hxx \
 	singleapp/qtlocalpeer.h \
 	singleapp/qtlockedfile.h \
 	singleapp/qtsingleapplication.h \
@@ -96,18 +78,6 @@ SOURCES	+= main.cpp \
 	manhattanstyle.cpp \
 	stylehelper.cpp \
 	styleanimator.cpp \
-	hunspell/affentry.cxx \
-	hunspell/affixmgr.cxx \
-	hunspell/csutil.cxx \
-	hunspell/dictmgr.cxx \
-	hunspell/hashmgr.cxx \
-	hunspell/hunspell.cxx \
-	hunspell/phonet.cxx \
-	hunspell/suggestmgr.cxx \
-	hunspell/utf_info.cxx \
-	hunspell/filemgr.cxx \
-	hunspell/replist.cxx \
-	hunspell/hunzip.cxx \
 	singleapp/qtlocalpeer.cpp \
 	singleapp/qtlockedfile.cpp \
 	singleapp/qtsingleapplication.cpp \
