From 1a249bfbc5775bdf7e38e909626c14dff20fc1b9 Mon Sep 17 00:00:00 2001
From: Hong Jen Yee (PCMan) <pcman.tw@gmail.com>
Date: Mon, 8 Nov 2010 04:02:06 +0800
Subject: [PATCH 1/1] List "gobject-2.0" in configure.ac explicitly.

---
 configure.ac |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index a9f5e05..790f7b6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,12 +43,12 @@ PKG_CHECK_MODULES(GTK, [$pkg_modules])
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
-glib_modules="glib-2.0 >= 2.18.0"
+glib_modules="glib-2.0 >= 2.18.0 gobject-2.0"
 PKG_CHECK_MODULES(GLIB, [$glib_modules])
 AC_SUBST(GLIB_CFLAGS)
 AC_SUBST(GLIB_LIBS)
 
-gio_modules="gthread-2.0 gio-unix-2.0 glib-2.0 >= 2.18.0"
+gio_modules="gthread-2.0 gio-unix-2.0 glib-2.0 >= 2.18.0 gobject-2.0"
 PKG_CHECK_MODULES(GIO, [$gio_modules])
 AC_SUBST(GIO_CFLAGS)
 AC_SUBST(GIO_LIBS)
-- 
1.7.0.1

