Index: libfm-0.1.14/src/Makefile.am
===================================================================
--- libfm-0.1.14.orig/src/Makefile.am	2010-12-02 00:06:29.000000000 +0100
+++ libfm-0.1.14/src/Makefile.am	2010-12-02 00:06:34.000000000 +0100
@@ -285,7 +285,6 @@
 	$(GIO_CFLAGS) \
 	-DGIO_MODULE_DIR=\"$(giomodulesdir)\"	\
 	-DGVFS_LOCALEDIR=\""$(localedir)"\"	\
-	-DG_DISABLE_DEPRECATED \
 	$(NULL)
 
 libgiofm_la_LDFLAGS = 	\
