--- plugins/configure.ac
+++ plugins/configure.ac
@@ -241,6 +241,8 @@
                                          AC_CHECK_LIB(avcodec, h264_decoder, [HAVE_H264_DECODER=yes], [HAVE_H264_DECODER=no])
                                          ;;
                    esac
+		   CPPFLAGS="${CPPFLAGS} -D__STDC_CONSTANT_MACROS=1"
+		   AC_SUBST(CPPFLAGS)
                    AC_SUBST(HAVE_H263P)
                    AC_SUBST(HAVE_MPEG4)
                    OPAL_LIBAVCODEC_HEADER
