--- sslstrip.py	2011-05-15 19:16:30.000000000 +0200
+++ sslstripb.py	2012-01-31 23:21:54.215996070 +0100
@@ -90,7 +90,7 @@
     (logFile, logLevel, listenPort, spoofFavicon, killSessions) = parseOptions(argv)
         
     logging.basicConfig(level=logLevel, format='%(asctime)s %(message)s',
-                        filename=logFile, filemode='w')
+                        filename=logFile, filemode='a')
 
     URLMonitor.getInstance().setFaviconSpoofing(spoofFavicon)
     CookieCleaner.getInstance().setEnabled(killSessions)
