--- src/defaults.c      2009-09-23 17:59:57.000000000 +0200
+++ src/defaults.c      2009-10-17 15:45:59.137361502 +0200
@@ -146,7 +146,7 @@
  * the visual guide for jaunty-notifications */
 #define DEFAULT_FADE_IN_TIMEOUT      250
 #define DEFAULT_FADE_OUT_TIMEOUT     1000
-#define DEFAULT_ON_SCREEN_TIMEOUT    10000
+#define DEFAULT_ON_SCREEN_TIMEOUT    4000
 
 /* GConf-keys to watch */
 #define GCONF_UI_FONT_NAME        "/desktop/gnome/interface/font_name"
