diff -urNp owanttask-1.1/src/org/objectweb/util/ant/JavadocMultipleLink.java owanttask-1.1.new/src/org/objectweb/util/ant/JavadocMultipleLink.java
--- owanttask-1.1/src/org/objectweb/util/ant/JavadocMultipleLink.java	2004-01-26 10:55:26.000000000 +0200
+++ owanttask-1.1.new/src/org/objectweb/util/ant/JavadocMultipleLink.java	2006-07-04 11:51:51.000000000 +0300
@@ -85,8 +85,8 @@ public class JavadocMultipleLink
 
 		// Add files
 		try {
-			for(Enumeration enum = v.elements(); enum.hasMoreElements();) {
-				File current = new File((File) enum.nextElement(),
+			for(Enumeration enum1 = v.elements(); enum1.hasMoreElements();) {
+				File current = new File((File) enum1.nextElement(),
 					TEMP_CLASS_NAME + ".java");
 				if (!current.createNewFile())
 					continue;
@@ -103,8 +103,8 @@ public class JavadocMultipleLink
 		super.execute();
 
 		// Remove files
-		for(Enumeration enum = v.elements(); enum.hasMoreElements();) {
-			new File((File) enum.nextElement(),
+		for(Enumeration enum1 = v.elements(); enum1.hasMoreElements();) {
+			new File((File) enum1.nextElement(),
 				TEMP_CLASS_NAME + ".java").delete();
 		}
 	}
